src/sig/sig.h

```diff
@@ -63,108 +84,123 @@ struct OQS_SIG {
+OQS_STATUS OQS_SIG_keypair(const OQS_SIG *sig, uint8_t *public_key, uint8_t *secret_key);
 
 /**
- * Generates a new signature.
- * @param s         Pointer to the signature object.
- * @param priv      Pointer to the signer's private key, of expected length `s->priv_key_len` bytes.
- * @param msg       Pointer to the message to sign.
- * @param msg_len   Length of the message to sign `msg`.
- * @param sig       Pointer where the generated signature will be stored. Caller must have allocated `s->max_sig_len` bytes.
- * @param sig_len   Pointer to the length of the generated signature.
- * @return          OQS_SUCCESS on success, or OQS_ERROR on failure
+ * Signature generation algorithm.
+ *
+ * Caller is responsible for allocating sufficient memory for `signed_message`,
+ * based on the `length_*` members in this object or the per-scheme
+ * compile-time macros `OQS_SIG_*_length_*`.
+ *
+ * @param[in] sig The OQS_SIG object representing the signature scheme.
+ * @param[out] signature The signature on the message represented as a byte string.
+ * @param[out] signature_len The length of the signature.
+ * @param[in] message The message to sign represented as a byte string.
+ * @param[in] message_len The length of the message to sign.
+ * @param[in] secret_key The secret key represented as a byte string.
+ * @return OQS_SUCCESS or OQS_ERROR
  */
-OQS_API OQS_STATUS OQS_SIG_sign(const OQS_SIG *s, const uint8_t *priv, const uint8_t *msg, const size_t msg_len, uint8_t *sig, size_t *sig_len);
+OQS_STATUS OQS_SIG_sign(const OQS_SIG *sig, uint8_t *signature, size_t *signature_len, const uint8_t *message, size_t message_len, const uint8_t *secret_key);
 
 /**
- * Verifies a signature.
- * @param s         Pointer to the signature object.
- * @param pub       Pointer to the signer's public key, of expected length `s->pub_key_len` bytes.
- * @param msg       Pointer to the signed message.
- * @param msg_len   Length of the signed message `msg`.
- * @param sig       Pointer to the signature.
- * @param sig_len   Length of the signature.
- * @return          OQS_SUCCESS on success, or OQS_ERROR on failure
+ * Signature verification algorithm.
+ *
+ * @param[in] sig The OQS_SIG object representing the signature scheme.
+ * @param[in] message The message represented as a byte string.
+ * @param[in] message_len The length of the message.
+ * @param[in] signature The signature on the message represented as a byte string.
+ * @param[in] signature_len The length of the signature.
+ * @param[in] public_key The public key represented as a byte string.
+ * @return OQS_SUCCESS or OQS_ERROR
  */
-OQS_API OQS_STATUS OQS_SIG_verify(const OQS_SIG *s, const uint8_t *pub, const uint8_t *msg, const size_t msg_len, const uint8_t *sig, const size_t sig_len);
+OQS_STATUS OQS_SIG_verify(const OQS_SIG *sig, uint8_t *message, size_t message_len, const uint8_t *signature, size_t signature_len, const uint8_t *public_key);
```

Comment:
`message, `message_len`, and `signature_len` should be marked as `const`

Reply:
Yes; leftovers from copy-pasting from nist-branch

Reply:
Well we do it for uint8_t which can't be changed by the function either, so why one and not the other?